README.md

```diff
@@ -1,62 +1,549 @@
+
+```hbs
+<HyperTableV2 @handler={{this.tableHandler}} @features={{this.features}} @options={{this.options}}>
+  {{! Custom search block (optional) }}
+  <:search>
+    <MyCustomSearch @onSearch={{this.handleSearch}} />
+  </:search>
+
+  {{! Contextual actions (optional) }}
+  <:contextual-actions>
+    <button {{on 'click' this.doSomething}}>Contextual Action</button>
+  </:contextual-actions>
+
+  {{! Table actions (optional) }}
+  <:table-actions>
+    <button {{on 'click' this.doSomething}}>Table Action</button>
+  </:table-actions>
+
+  {{! Custom empty state (optional) }}
+  <:empty-state>
+    <div class='custom-empty-state'>This is an empty state</div>
+  </:empty-state>
+</HyperTableV2>
+```
+
+### 3. Interface Implementations
+
+#### TableManager
+
+```typescript
+class MyTableManager implements TableManager {
+  async fetchColumnDefinitions(): Promise<{ column_definitions: ColumnDefinition[] }> {
+    // Fetch available column definitions from API
+    const response = await fetch('/api/column-definitions');
+    return response.json();
+  }
+
+  async fetchColumns(): Promise<{ columns: Column[] }> {
+    // Fetch current column configuration from API
+    const response = await fetch('/api/columns');
+    return response.json();
+  }
+
+  async upsertColumns(request: { columns: Column[] }): Promise<{ columns: Column[] }> {
+    // Save column configuration to API
+    const response = await fetch('/api/columns', {
+      method: 'POST',
+      body: JSON.stringify(request)
+    });
+    return response.json();
+  }
+
+  // Optional: for faceted filtering
+  async fetchFacets(columnKey: string, filteringKey: string, searchValue?: string): Promise<FacetsResponse> {
+    const response = await fetch(`/api/facets/${columnKey}?filtering_key=${filteringKey}&search=${searchValue}`);
+    return response.json();
+  }
+}
+```
+
+**Available methods:**
+
+- `fetchColumnDefinitions()` - Returns available column definitions
+- `fetchColumns()` - Returns current column configuration
+- `upsertColumns(request)` - Saves column configuration
+- `fetchFacets(columnKey, filteringKey, searchValue)` - Optional: provides faceted filtering
+
+#### RowsFetcher
+
+```ts
+class MyRowsFetcher implements RowsFetcher {
+  async fetch(page: number, perPage: number): Promise<{ rows: Row[]; meta: { total: number } }> {
+    // Fetch paginated data from API
+    const response = await fetch(`/api/data?page=${page}&per_page=${perPage}`);
+    return response.json();
+  }
+
+  // Optional: for individual row updates
+  async fetchById(recordId: number): Promise<Row> {
+    const response = await fetch(`/api/data/${recordId}`);
+    return response.json();
+  }
+}
+```
+
+**Available methods:**
+
+- `fetch(page, perPage)` - Returns paginated data
+- `fetchById(id)` - Optional: fetches individual rows
+
+#### Custom rendering resolver (Optional)
+
+If no RenderingResolver is provided, Hypertable uses a default resolver with built-in renderers.
```

Comment:
[...] with built-in renderers that support:

- `integer` type: this will properly format numbers in cells and brings range-based filtering
- `timestamp` type: properly displays dates and brings calendar-based filtering
- a default `text` renderer to display textual content and filter using a typeahead input